### PR TITLE
Issue #7208: aligned javadoc/xdoc for SingleLineJavadoc

### PR DESCRIPTION
**Pull request by pbludov**
Issue #7208 (part of #5750)

No changes in the code.

### CODE REVIEW

**Comment by romani** [comment]
In separate PR, please add your self to https://github.com/checkstyle/checkstyle/blob/master/pom.xml#L48, 
@rnveach , I pretty sure that @pbludov, did major impact on project and he absolutely ready to be granted RW access. Please confirm.

**Comment by romani** [comment]
@pbludov, is it a last PR for both issues ?

**Comment by pbludov** [comment]
> @pbludov, is it a last PR for both issues ?

yes, this is the last one.

File: src/test/java/com/puppycrawl/tools/checkstyle/internal/XdocsJavaDocsTest.java [hunk 1]
```diff
@@ -64,22 +63,10 @@
     private static final Map<String, String> CHECK_PROPERTY_DOC = new HashMap<>();
     private static final Map<String, String> CHECK_TEXT = new HashMap<>();
 
-    /**
-     * The list of checks that are not yet compatible with this rule.
-     */
-    private static final String[] INCOMPATIBLE_CHECKS = {
```
Comment (pbludov) [inline src/test/java/com/puppycrawl/tools/checkstyle/internal/XdocsJavaDocsTest.java]: No need for this property anymore.

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/SummaryJavadocCheck.java [hunk 2]
```diff
@@ -97,7 +97,7 @@
  * </p>
  * <pre>
  * &lt;module name="SummaryJavadocCheck"&gt;
- *   &lt;property name="period" value="&#12290;"/&gt;
```
Comment (pbludov) [inline src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/SummaryJavadocCheck.java]: This change to fix https://bugs.openjdk.java.net/browse/JDK-8148480

Comment (romani) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/SummaryJavadocCheck.java]: it is addressed in https://github.com/checkstyle/checkstyle/pull/7266

**Comment by rnveach** [comment]
> @rnveach , I pretty sure that @pbludov, did major impact on project and he absolutely ready to be granted RW access. Please confirm.

I confirm.

**Comment by romani** [comment]
please rebase and resolve conflicts

**Comment by pbludov** [comment]
> please rebase and resolve conflicts

done

**Comment by pbludov** [comment]
> Not sure why escaping is required

Done. You are right: no escaping need for xdocs. The check `XdocsJavaDocsTest` is smart enough to compare escaped symbols on the Java side with plain symbols on xdocs side.

**Comment by romani** [comment]
I still see escaping in code diff, please recheck that you pushed sources.

**Comment by pbludov** [comment]
> I still see escaping in code diff, please recheck that you pushed sources.

I did. Escaped characters are now only in the Java file.

**Pull request merged**